testsuite/pytests/sli2py_regressions/test_issue_659.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+#
+# test_issue_659.py
+#
+# This file is part of NEST.
+#
+# Copyright (C) 2004 The NEST Initiative
+#
+# NEST is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# NEST is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+"""
+Regression test for Issue #659 (GitHub).
+
+This test checks that calling Run or Cleanup without calling Prepare first results in an error.
+"""
+
+import nest
+import pytest
+
+
+def test_run_without_prepare():
+    """
+    Test that calling Run without Prepare results in an error.
+    """
+    nest.ResetKernel()
+    with pytest.raises(Exception):
```

Comment:
We can and should be more specific here:
```suggestion
    with pytest.raises(nest.kernel.NESTError):
```
This applies also to the remaining cases.